tracing/src/main/java/com/palantir/remoting1/tracing/AsyncSlf4jSpanObserver.java

```diff
@@ -28,6 +35,44 @@
 
     private final Logger logger;
 
+    @JsonDeserialize(as = ImmutableZipkinCompatibleSerializableSpan.class)
+    @JsonSerialize(as = ImmutableZipkinCompatibleSerializableSpan.class)
+    @Value.Immutable
+    @Value.Style(visibility = Value.Style.ImplementationVisibility.PACKAGE)
+    abstract static class ZipkinCompatibleSerializableSpan {
+        private static final ObjectMapper mapper = new ObjectMapper().registerModule(new GuavaModule());
+
+        abstract String getTraceId();
+        abstract String getId();
+        abstract String getName();
+        abstract Optional<String> getParentId();
+        abstract long getTimestamp();
+        abstract long getDuration();
+
+        static ZipkinCompatibleSerializableSpan fromSpan(Span span) {
+            return ImmutableZipkinCompatibleSerializableSpan.builder()
+                    .traceId(span.getTraceId())
+                    .id(span.getSpanId())
+                    .name(span.getOperation())
+                    .parentId(span.getParentSpanId())
+                    .timestamp(span.getStartTimeMicroSeconds())
+                    .duration(nanoToMicro(span.getDurationNanoSeconds()))  // Zipkin-durations are micro-seconds, rounded up
+                    .build();
+        }
+
+        static long nanoToMicro(long nano) {
+            return (long) Math.ceil(nano / (double) 1000);
```

Comment:
Why are we doing double math here? This should be:

`return Math.max(1L, nano / 1000L);`